### PR TITLE
Itemviews refactoring

### PR DESCRIPTION
**Pull request by zas**
# Summary

* This is a…
  * [ ] Bug fix
  * [ ] Feature addition
  * [x] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

* JIRA ticket (_optional_): PICARD-XXX

# Solution

# Action

Additional actions required:
* [ ] Update Picard [documentation](https://github.com/metabrainz/picard-docs) (please include a reference to this PR)
* [ ] Other (please specify below)

### CODE REVIEW

File: picard/ui/itemviews.py [hunk 1]
```diff
@@ -337,64 +451,47 @@ def select_object(self, obj):
                 break
 
 
-def paint_column_icon(painter, rect, icon):
+def paint_column_icon(painter, rect, column):
```
Comment (phw) [inline picard/ui/itemviews.py:454]: The history of this function is that it originally was used for painting both the header and the content of the cells, but it was using different icons (gray in the header, gray or colored in the columns). This was later changed to use `TreeItem.setIcon()` in the content cells for performance reason (even though it causes a slight misalignment, but it is much faster than a custom paint).

Anyway, my suggestion would be to move the `paint_column_icon` to `IconColumn.paint_header(painter, rect)`, as it now is tightly couple to the IconColumn class anyway and also is only about painting the header. Maybe also rename `IconColumn.icon` to `IconColumn.header_icon`.

Comment (zas) [reply picard/ui/itemviews.py]: Done in https://github.com/metabrainz/picard/pull/2454/commits/5fb55d30e0fccafbfc58f11dc926a0ae9ed35e65 and https://github.com/metabrainz/picard/pull/2454/commits/bc99b478b8a2715a1db50d78b810a864108c4df9

File: picard/extension_points/__init__.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
```
Comment (phw) [inline picard/extension_points/__init__.py:1]: What about putting this into a separate file `extension_points/item_actions.py`? I fear if we start putting this into `__init__.py` then we will end up with all different extension points being shoved into this file.

Comment (zas) [reply picard/extension_points/__init__.py:1]: Done in https://github.com/metabrainz/picard/pull/2454/commits/e11d2a6e68fd5233032137b054d85969bcd5f12b (and rebased)

**Comment by zas** [comment]
Obsolete PR, most changes were introduced in smaller PRs.

See:
https://github.com/metabrainz/picard/pull/2461
https://github.com/metabrainz/picard/pull/2462
https://github.com/metabrainz/picard/pull/2463
https://github.com/metabrainz/picard/pull/2464
https://github.com/metabrainz/picard/pull/2466

**Pull request closed without merging**